src/PrestaShopBundle/Command/ConfigCommand.php

```diff
@@ -0,0 +1,320 @@
+            throw new Exception($msg, self::STATUS_FAILED_SHOPCONSTRAINT);
+        }
+    }
+
+    /**
+     * initialize language if the option was given
+     */
+    private function initLanguage(): void
+    {
+        $inputlang = $this->input->getOption('lang');
+        if (!$inputlang) {
+            return;
+        }
+
+        // all languages
+        $onlyActive = true;
+        $onlyShopId = null;
+        if (!is_null($this->shopConstraint->getShopId())) {
+            $onlyShopId = $this->shopConstraint->getShopId()->getValue();
+        }
+        $languages = $this->languageDataProvider->getLanguages($onlyActive, $onlyShopId);
+
+        $found = current(array_filter($languages, function (array $item) use ($inputlang) {
+            $key = is_numeric($inputlang) ? 'id_lang' : 'iso_code';
+
+            return isset($item[$key]) && $inputlang == $item[$key];
+        }));
+
+        if (!$found) {
+            throw new Exception('Invalid language', self::STATUS_INVALID_OPTIONS);
+        }
+
+        $this->idLang = (int) $found['id_lang'];
+    }
+
+    /**
+     * Get and print out one configuration value
+     */
+    private function get(): void
+    {
+        $key = $this->input->getArgument('key');
+
+        if (strpos($key, '*') !== false) {
+            if (!$this->idLang) {
+                throw new Exception('Using wildcards requires --lang option', self::STATUS_LANG_REQUIRED);
+            }
+            try {
+                $keys = array_filter($this->configuration->keys(), function (string $v) use ($key) {
+                    return fnmatch($key, $v);
+                });
+            } catch (NotImplementedException $e) {
+                throw new Exception('Not implemented yet', self::STATUS_NOT_IMPLEMENTED);
+            } catch (Exception $e) {
+                $msg = sprintf('Could not load all configuration keys: %s', $e->getMessage());
+                throw new Exception($msg, self::STATUS_ERROR);
+            }
+        } else {
+            $keys = [$key];
+        }
+
+        foreach ($keys as $key) {
+            $value = $this->configuration->get($key, null, $this->shopConstraint);
+
+            // non language values will be just strings
+            // but for language dependant values the response is an array
+            if (is_array($value)) {
+                if (!$this->idLang) {
+                    $msg = sprintf('%s is language dependant, --lang option is required', $key);
+                    throw new Exception($msg, self::STATUS_LANG_REQUIRED);
+                }
+                $value = $value[$this->idLang];
+            }
+
+            $this->io->success(sprintf('%s=%s', $key, $value));
+        }
+    }
+
+    /**
+     * Set and print out one configuration value
+     */
+    private function set(): void
+    {
+        $key = $this->input->getArgument('key');
+
+        if (strpos($key, '*') !== false) {
```

Comment:
Why this test? We should ignore `*` and consider it as a string only
```suggestion
        if (Validate::isConfigName($key) === false) {
```

Reply:
`*` acts as a wildcard. When configuration->keys someday implements the "get all keys" (here: https://github.com/PrestaShop/PrestaShop/blob/901a073a6466642d20bc6440e83fc472bfa7ceb5/src/Adapter/Configuration.php#L69) one can do this

```
bin/console prestashop:config get mymodule_* --lang fi
```

and get a list of all the configration values that match `mymodule_*`. Very useful when you are checking what a module does (as long as it has prefixed its config values properly)

Reply:
That's what I said, we are in a `set` context, to `get` you can remove it here.

Reply:
If you accept Pierre's suggestion you could also change the error name, to the one you have in another method:

> throw new Exception('"' . $key . '" is not a valid configuration key', self::STATUS_INVALID_OPTIONS);